text/0038-output-types-for-tokenization-and-sc/0038-output-types-for-tokenization-and-sc.md

```diff
@@ -0,0 +1,2994 @@
+<table>
+    <tr>
+        <td><b>Name</b></td>
+        <td><b>Type</b></td>
+        <td><b>Description</b></td>
+    </tr>
+    <tr>
+        <td>Block Type</td>
+        <td>uint8</td>
+        <td>
+            Set to <strong>value 6</strong> to denote a <i>Expiration Unix Block</i>.
+        </td>
+    </tr>
+    <tr>
+        <td>Unix Time</td>
+        <td>uint32</td>
+        <td>
+            Before this unix time (seconds since Unix epoch), <i>Address</i> is allowed to unlock the output, after that only the address defined in <i>Sender Block</i>.
+        </td>
+    </tr>
+</table>
+
+#### Metadata Block
+
+Outputs may carry additional data with them that is interpreted by higher layer applications built on the Tangle. The
+protocol treats this metadata as pure binary data, it has no effect on the validity of an output except that it
+increases the required dust deposit. ISCP is a great example of a higher layer protocol that makes use of
+<i>Metadata Block</i>: smart contract request parameters are encoded in the metadata field of outputs.
+
+##### Additional syntactic transaction validation rules:
+- An output with <i>Metadata Block</i> is valid, if and only if 0 < `Data Length` ≤ `Max Metadata Length`.
+
+<details>
+<summary>Metadata Block</summary>
+<blockquote>
+    Defines metadata (arbitrary binary data) that will be stored in the output.
+</blockquote>
+</details>
+
+<table>
+    <tr>
+        <td><b>Name</b></td>
+        <td><b>Type</b></td>
+        <td><b>Description</b></td>
+    </tr>
+    <tr>
+        <td>Block Type</td>
+        <td>uint8</td>
+        <td>
+            Set to <strong>value 7</strong> to denote a <i>Metadata Block</i>.
+        </td>
+    </tr>
+    <tr>
+        <td>Data Length</td>
+        <td>uint32</td>
+        <td>
+            Length of the following data field in bytes.
+        </td>
+    </tr>
+    <tr>
+        <td>Data</td>
+        <td>ByteArray</td>
+        <td>Binary data.</td>
+    </tr>
+</table>
+
+#### Indexation Block
+
+An indexation block makes it possible to tag outputs with an index, so they can be retrieved through an API not only by
+their address, but also based on the the `Indexation Tag`. **The combination of an <i>Indexation Block</i>, a
+<i>Metadata Block</i> and a <i>Sender Block</i> makes it possible to retrieve data associated to an address and stored
+in outputs that was created by a specific party (`Sender`) for a specific purpose (`Indexation Tag`).**
+
+Storing indexed data in outputs however incurs greater dust deposit for such outputs, because they create look-up
+entries in nodes' databases.
+
+##### Additional syntactic transaction validation rules:
+- An output with <i>Indexation Block</i> is valid, if and only if 0 < `Indexation Tag Length` ≤
+  `Max Indexation Tag Length`.
+
+<details>
+<summary>Indexation Block</summary>
+<blockquote>
+    Defines an indexation tag to which the <i>Metadata Block</i> will be indexed. Creates an indexation lookup in nodes.
```

Comment:
Why specify `Metadata Block` ? There is not necessarily one.

Reply:
Right, it is the output that is indexed anyway. Whether it has metadata block or not doesn't matter.